nixarr/recyclarr/default.nix

```diff
@@ -0,0 +1,128 @@
+{ config
+, lib
+, pkgs
+, inputs
+, ...
+}:
+with lib; let
+  cfg = config.nixarr.recyclarr;
+  nixarr = config.nixarr;
+
+  # Helper function to extract API keys
+  extractApiKeys = pkgs.writeShellApplication {
+    name = "extract-recyclarr-api-keys";
+    runtimeInputs = with pkgs; [ yq ];
+    text = ''
+      # Ensure state directory exists with proper permissions
+      mkdir -p "${cfg.stateDir}"
+      chown ${config.services.recyclarr.user}:${config.services.recyclarr.group} "${cfg.stateDir}"
+      chmod 755 "${cfg.stateDir}"
+
+      ${optionalString nixarr.radarr.enable ''
+        # Extract Radarr API key
+        API_KEY_FILE="${cfg.stateDir}/radarr-api-key"
+        xq -r '.Config.ApiKey' "${nixarr.radarr.stateDir}/config.xml" > "$API_KEY_FILE"
+        chmod 400 "$API_KEY_FILE"
+        chown ${config.services.recyclarr.user}:${config.services.recyclarr.group} "$API_KEY_FILE"
+        echo "RADARR_API_KEY=$(tr -d '\n' < "$API_KEY_FILE")" >> "${cfg.stateDir}/env"
+      ''}
+
+      ${optionalString nixarr.sonarr.enable ''
+        # Extract Sonarr API key
+        API_KEY_FILE="${cfg.stateDir}/sonarr-api-key"
+        xq -r '.Config.ApiKey' "${nixarr.sonarr.stateDir}/config.xml" > "$API_KEY_FILE"
+        chmod 400 "$API_KEY_FILE"
+        chown ${config.services.recyclarr.user}:${config.services.recyclarr.group} "$API_KEY_FILE"
+        echo "SONARR_API_KEY=$(tr -d '\n' < "$API_KEY_FILE")" >> "${cfg.stateDir}/env"
+      ''}
+
+      chmod 400 "${cfg.stateDir}/env"
+      chown ${config.services.recyclarr.user}:${config.services.recyclarr.group} "${cfg.stateDir}/env"
+    '';
+  };
+in {
+  options.nixarr.recyclarr = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      example = true;
+      description = ''
+        Whether or not to enable the Recyclarr service. This service does not need to be run behind a VPN.
+
+        **Required options:** [`nixarr.enable`](#nixarr.enable)
+      '';
+    };
+
+    package = mkPackageOption pkgs "recyclarr" { };
+
+    stateDir = mkOption {
+      type = types.path;
+      default = "${nixarr.stateDir}/recyclarr";
+      defaultText = literalExpression ''"''${nixarr.stateDir}/recyclarr"'';
+      example = "/nixarr/.state/recyclarr";
+      description = "The location of the state directory for the Recyclarr service.";
+    };
+
+    configFile = mkOption {
+      type = types.path;
```

Comment:
This should probably just be an attribute set, and then converted into yaml. Then a user wouldn't have to write a seperate yaml file, and could just write `recyclarr.config.sonarr.base_url = "http://localhost:8989"`.

Reply:
Yea, I wasn't sure if that would support the `!env_var` macro. In many cases you will find provided profiles, and they're all in plain YAML (eg. TRaSH Guide provides YAML examples). Maybe we could offer both? [Upstream uses attrsets](https://search.nixos.org/options?channel=unstable&show=services.recyclarr.configuration&from=0&size=50&sort=relevance&type=packages&query=recyclarr).

Reply:
I actually like both, it would be "unclean" for nixpkgs, but Nixarr is meant to be more specialized and opinionated so it could work. Just make sure that the options are mutually exclusive and document the fact.

Reply:
Added it. I also tried mapping the yaml file to the new attrset option, but no native YAML support in nix, and `readJSON` bails if the YAML contains comments.

Reply:
I think this is fine